commit c136b84393d4e340e1b53fc7f737dd5827b19ee5
Merge: e0f25a3f2d05 1372324b328c
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Jul 6 18:38:31 2017 -0700

    Merge tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm
    
    Pull KVM updates from Paolo Bonzini:
     "PPC:
       - Better machine check handling for HV KVM
       - Ability to support guests with threads=2, 4 or 8 on POWER9
       - Fix for a race that could cause delayed recognition of signals
       - Fix for a bug where POWER9 guests could sleep with interrupts pending.
    
      ARM:
       - VCPU request overhaul
       - allow timer and PMU to have their interrupt number selected from userspace
       - workaround for Cavium erratum 30115
       - handling of memory poisonning
       - the usual crop of fixes and cleanups
    
      s390:
       - initial machine check forwarding
       - migration support for the CMMA page hinting information
       - cleanups and fixes
    
      x86:
       - nested VMX bugfixes and improvements
       - more reliable NMI window detection on AMD
       - APIC timer optimizations
    
      Generic:
       - VCPU request overhaul + documentation of common code patterns
       - kvm_stat improvements"
    
    * tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm: (124 commits)
      Update my email address
      kvm: vmx: allow host to access guest MSR_IA32_BNDCFGS
      x86: kvm: mmu: use ept a/d in vmcs02 iff used in vmcs12
      kvm: x86: mmu: allow A/D bits to be disabled in an mmu
      x86: kvm: mmu: make spte mmio mask more explicit
      x86: kvm: mmu: dead code thanks to access tracking
      KVM: PPC: Book3S: Fix typo in XICS-on-XIVE state saving code
      KVM: PPC: Book3S HV: Close race with testing for signals on guest entry
      KVM: PPC: Book3S HV: Simplify dynamic micro-threading code
      KVM: x86: remove ignored type attribute
      KVM: LAPIC: Fix lapic timer injection delay
      KVM: lapic: reorganize restart_apic_timer
      KVM: lapic: reorganize start_hv_timer
      kvm: nVMX: Check memory operand to INVVPID
      KVM: s390: Inject machine check into the nested guest
      KVM: s390: Inject machine check into the guest
      tools/kvm_stat: add new interactive command 'b'
      tools/kvm_stat: add new command line switch '-i'
      tools/kvm_stat: fix error on interactive command 'g'
      KVM: SVM: suppress unnecessary NMI singlestep on GIF=0 and nested exit
      ...

diff --cc arch/s390/kvm/gaccess.c
index 875f8bea8c67,17e3a4e71bc9..653cae5e1ee1
--- a/arch/s390/kvm/gaccess.c
+++ b/arch/s390/kvm/gaccess.c
@@@ -790,8 -808,8 +808,8 @@@ static int guest_page_range(struct kvm_
  			return trans_exc(vcpu, PGM_PROTECTION, ga, ar, mode,
  					 PROT_TYPE_LA);
  		ga &= PAGE_MASK;
 -		if (psw_bits(*psw).t) {
 +		if (psw_bits(*psw).dat) {
- 			rc = guest_translate(vcpu, ga, pages, asce, mode);
+ 			rc = guest_translate(vcpu, ga, pages, asce, mode, &prot);
  			if (rc < 0)
  				return rc;
  		} else {
@@@ -899,10 -918,10 +918,10 @@@ int guest_translate_address(struct kvm_
  					 mode, PROT_TYPE_LA);
  	}
  
 -	if (psw_bits(*psw).t && !asce.r) {	/* Use DAT? */
 +	if (psw_bits(*psw).dat && !asce.r) {	/* Use DAT? */
- 		rc = guest_translate(vcpu, gva, gpa, asce, mode);
+ 		rc = guest_translate(vcpu, gva, gpa, asce, mode, &prot);
  		if (rc > 0)
- 			return trans_exc(vcpu, rc, gva, 0, mode, PROT_TYPE_DAT);
+ 			return trans_exc(vcpu, rc, gva, 0, mode, prot);
  	} else {
  		*gpa = kvm_s390_real_to_abs(vcpu, gva);
  		if (kvm_is_error_gpa(vcpu->kvm, *gpa))